app/services/valid_test_data_generators/mentor_ect_generator.rb

```diff
@@ -51,13 +51,17 @@ def schedule
 
     def find_or_create_school_cohort
       school = partnership.school
-      school_cohort = SchoolCohort.find_or_create_by!(school:, cohort:, induction_programme_choice: "full_induction_programme")
-      InductionProgramme.find_or_create_by!(school_cohort:, partnership:, training_programme: "full_induction_programme")
+      school_cohort = SchoolCohort.find_or_create_by!(school:, cohort:) do |sc|
```

Comment:
What's the reason we are only matching on school/cohort but set programme choice to FIP on creation if it doesn't exist?

Reply:
Because I was having issues on that line because there's a unique constraint in school & cohort..

`SchoolCohort.find_or_create_by!(school:, cohort:, induction_programme_choice: "full_induction_programme")` might try to create a new record for the same school & cohort if `induction_programme_choice` is other than `full_induction_programme` in an existing record.

Reply:
Ah I see, that makes sense thanks 👍